### DISCUSSION TITLE
UTF Issue when shortening a URL with https

### DISCUSSION THREAD

**Discussion by og-shg** [YOURLS/YOURLS#3707] [General]
If I shorten this URL
https://search.spokanehomeguy.com/search/detail/202195346?s[page]=1&s[locations][0][city]=Spokane&s[locations][0][state]=WA

The script spins and spins and does not produce an output.  

If I remove the "s" 
http://search.spokanehomeguy.com/search/detail/202195346?s[page]=1&s[locations][0][city]=Spokane&s[locations][0][state]=WA

It shortens just fine.  
How can I fix this without constantly adjusting the input?

Error Log:
[18-Jan-2024 22:48:28 UTC] PHP Fatal error:  Uncaught ValueError: mb_convert_encoding(): Argument #3 ($from_encoding) contains invalid encoding "utf" in /home/fbvvkrxo4bco/public_html/includes/functions.php:933
Stack trace:
#0 /home/fbvvkrxo4bco/public_html/includes/functions.php(933): mb_convert_encoding()
#1 /home/fbvvkrxo4bco/public_html/includes/functions-shorturls.php(97): yourls_get_remote_title()
#2 /home/fbvvkrxo4bco/public_html/admin/admin-ajax.php(21): yourls_add_new_link()
#3 {main}
  thrown in /home/fbvvkrxo4bco/public_html/includes/functions.php on line 933

**Comment by dgw** [comment]
The markup served by this URL is not valid. Specifically, this is an error:

```html
<meta charSet="utf=8"/>
```

Note the `=` inside the value, which should be a `-`. https://developer.mozilla.org/en-US/docs/Web/HTML/Element/meta#charset

I'm not positive why there's a difference between `http://` and `https://`, but they seem to do some user-agent sniffing to decide whether to redirect the client or not, e.g. a browser is redirected from `http://` to `https://` but `curl` instead gets a very brief "Hello Developer Person!" page with a comment asking to use the HTTPS link directly.

The fatal error comes from inside YOURLS's attempt to convert an unrecognized encoding interpreted as `utf` into proper `utf-8`:

https://github.com/YOURLS/YOURLS/blob/be573de7434f62954eaf15fde636466dc158e711/includes/functions.php#L911-L938

YOURLS should probably catch this error in case of an invalid charset value and try `mb_convert_encoding()` on the title again without specifying the source charset. The title potentially being mangled if the charset is incorrectly converted should be less of a problem than outright failing on invalid charset values as it does now.

~~Momentarily, I will create an issue from this discussion to track the proposed enhancement.~~ See #3708

**Reply by og-shg** [reply]
Thanks so much, that would do it.  I did submit a ticket to our vendor too to fix the problem on our end too.   Thank you for your prompt response.
